src/hooks/useAuthenticated.ts

```diff
@@ -22,20 +22,18 @@ const useAuthenticated = (): HooksReturnType => {
         profilePicture: DEFAULT_AVATAR,
     });
 
+    const { data, isUserAuthorized } = useUserData();
     const [isLoading, setIsLoading] = useState(false);
-
     useEffect(() => {
         const fetchData = async () => {
             setIsLoading(true);
             try {
-                const { requestPromise } = fetch({ url: USER_SELF });
-                const { data } = await requestPromise;
-                if (data.incompleteUserDetails) {
+                if (data?.incompleteUserDetails) {
                     window.open(`${SIGNUP_LINK}`, '_blank', 'noopener');
                 }
                 setUserData({
-                    userName: data.username,
-                    firstName: data.first_name,
+                    userName: data?.username,
+                    firstName: data?.first_name,
                     profilePicture: data?.picture?.url,
                 });
                 setIsLoggedIn(true);
```

Comment:
Your breaking the functionality of code by setting isLoggedIn to true no matter if the user is authenticated or not

Reply:
How come?

Reply:
How come?